src/app/qgisapp.cpp

```diff
@@ -3376,9 +3377,16 @@ void QgisApp::refreshProfileMenu()
     }
     else
     {
-      connect( action, &QAction::triggered, this, [this, name]()
+      connect( action, &QAction::triggered, this, [this, name, activeName]()
       {
+        // Launch a new instance of QGIS with the selected profile
         userProfileManager()->loadUserProfile( name );
+
+        // We want the actions to be checkable so the current profile is shown
+        // as checked, but we don't want to check the clicked action, so we
+        // check again the action linked to the active profile
+        findChild<QAction *>( "mActionProfile_" + activeName )->setChecked( true );
```

Comment:
Would 
```suggestion
        whileBlocking( action )->setChecked( true );
```

work instead?

Reply:
Not realy. `action` is the QAction of the userProfile we want to open. When we trigger it, it becomes checked in the current instance of Qgis, the current profile action becomes unchecked, and a new QGIS instance opens. We then recheck the current profile action (actions are in an exclusive QActionGroup). I don't see how blocking signals would help in this case?